src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,127 +135,361 @@ internal abstract class AbstractAddParameterCodeFixProvider<
-                if (constructor.IsNonImplicitAndFromSource() &&
-                    NonParamsParameterCount(constructor) < arguments.Count)
+                if (method.IsNonImplicitAndFromSource())
                 {
-                    var argumentToAdd = DetermineFirstArgumentToAdd(
-                        semanticModel, syntaxFacts, comparer, constructor,
-                        arguments, argumentOpt);
+                    var isNamedArgument = !string.IsNullOrWhiteSpace(syntaxFacts.GetNameForArgument(argumentOpt));
 
-                    if (argumentToAdd != null)
+                    if (isNamedArgument || NonParamsParameterCount(method) < arguments.Count)
                     {
-                        if (argumentOpt != null && argumentToAdd != argumentOpt)
+                        var argumentToAdd = DetermineFirstArgumentToAdd(
+                            semanticModel, syntaxFacts, comparer, method,
+                            arguments, argumentOpt);
+
+                        if (argumentToAdd != null)
                         {
-                            // We were trying to fix a specific argument, but the argument we want
-                            // to fix is something different.  That means there was an error earlier
-                            // than this argument.  Which means we're looking at a non-viable 
-                            // constructor.  Skip this one.
-                            continue;
+                            if (argumentOpt != null && argumentToAdd != argumentOpt)
+                            {
+                                // We were trying to fix a specific argument, but the argument we want
+                                // to fix is something different.  That means there was an error earlier
+                                // than this argument.  Which means we're looking at a non-viable 
+                                // constructor or method.  Skip this one.
+                                continue;
+                            }
+
+                            methodsAndArgumentToAdd.Add(new ArgumentInsertPositionData<TArgumentSyntax>(
+                                method, argumentToAdd, arguments.IndexOf(argumentToAdd)));
                         }
+                    }
+                }
+            }
+
+            return methodsAndArgumentToAdd.ToImmutableAndFree();
+        }
+
+        private int NonParamsParameterCount(IMethodSymbol method)
+            => method.IsParams() ? method.Parameters.Length - 1 : method.Parameters.Length;
+
+        private void RegisterFixForMethodOverloads(
+            CodeFixContext context,
+            SeparatedSyntaxList<TArgumentSyntax> arguments,
+            ImmutableArray<ArgumentInsertPositionData<TArgumentSyntax>> methodsAndArgumentsToAdd)
+        {
+            var codeFixData = PrepareCreationOfCodeActions(context.Document, arguments, methodsAndArgumentsToAdd);
+            var fixes = codeFixData.Length <= 2
```

Comment:
the code here could use a short paragraph explaining the overall experience we're trying to deliver here.